### PR TITLE
List of local searches

### LINKED ISSUE #44: Provide a list of recent local searches for the user to select from when interacting with the search box
We would like users to be able to select, from a dropdown below the top navigation search box, recent searches.

image

image

* The search terms should be stored in local storage and therefore no backend support is required.
* Initially, when there is no search term entered, the last n searches should be displayed
* Once there is a search term (or part of a term), perform an "includes" filter on the list and only display the most recent matching n searches. Extra points if we can highlight the match within the search.
* This should work _without_ the ClojureScript app and therefore the solution to be written in JavaScript (there are plenty of examples of this within the project)
* Extra points for allowing the searches to be deleted

### PR DESCRIPTION
**Pull request by ChaituVR**
Closes #44 

Still to do: Highlight search string in dropdown

Screenshots:
image
image

### CODE REVIEW

**Comment by acron0** [comment]
@ChaituVR Amazing work, thank you for this. Before I review it would you mind condensing some of the spread out formatting?
e.g. image

It's more canonical to keep things compact

**Comment by acron0** [comment]
image
Also getting an error when attempting to use the search box after a page refresh.

**Comment by ChaituVR** [comment]
Will do soon 👍

**Comment by ChaituVR** [comment]
@acron0 sorry for delay, fixed that and also added bold to mark typed in value

File: common/src/wh/components/navbar/components.cljc [hunk 1]
```diff
@@ -36,6 +39,42 @@
...
+        (let [values (remove (fn [x] (= x text)) local-search)]
+          (.setItem js/localStorage "local_search" 
+            (clojure.string/join "||" values) 
+          )
+          (dispatch [:wh.components.navbar.events/set-local-search values])
+        ))
+      ))
+    )]
+    
+  ]])
```
Comment (acron0) [inline common/src/wh/components/navbar/components.cljc:67]: This style, of breaking out the parentheses, is not idiomatic and not inline with the style of the rest of the code base. Please condense this. Thanks

**Review by acron0 — CHANGES_REQUESTED** [review]

File: common/src/wh/components/navbar/components.cljc (hunk 1 above)
Comment (ChaituVR) [reply common/src/wh/components/navbar/components.cljc]: @acron0 fixed those and also added background color to entered value 
image

**Comment by acron0** [comment]
Hi @ChaituVR we will look at this during this week, thanks a gain!

**Comment by jonki** [comment]
@ChaituVR appreciate your interest in this issue. Having said that there are few problems here. 

Our issue description says: "This should work without the ClojureScript app and therefore the solution to be written in JavaScript (there are plenty of examples of this within the project)". Although our application codebase is mostly Clojure/ClojureScript we do implement some parts of client app in pure JavaScript. This way we can give our users some basic interactivity without loading whole ClojureScript app. That is also the goal in this issue. This is not connected closely with rest of the application logic so we want to have this functionality implemented in pure JavaScript, not as part of the ClojureScript application.

Second thing: even if we'd like to use ClojureScript solution I see that you're a beginner Clojurist. Your code is ok as first approaches to Clojure, it's a bit too chaotic too be merge into our codebase. You're using unnecessary nested `let` forms, too many parens in some places, few maps for event handlers instead of one, etc… We cannot really start to discuss logic of your code, when code itself is too hairy.

If you're keen to continue your work on this issue (which we encourage you too!) please read our description of the issue again, thoroughly, and then ask all the questions you need to fully understand what we're asking for. We will respond in nearest possible future.

Take care, thank you so much for your engagement 🤝

**Pull request closed without merging**